app/controllers/candidate_interface/candidate_interface_controller.rb

```diff
@@ -38,17 +38,17 @@ def check_cookie_preferences
     # Should the current request be considered as made under the Your
     # applications tab
     def choices_controller?
-      @choices_controller ||= if current_application.continuous_applications?
-                                choices_controllers = Regexp.compile(APPLICATION_CHOICE_CONTROLLER_PATHS.join('|'))
+      return false if current_application.v23?
 
-                                if controller_path.match?(choices_controllers)
-                                  true
-                                elsif controller_path.match('candidate_interface/guidance')
-                                  request.referer&.match?('choices')
-                                end
-                              else
-                                false
-                              end
+      @choices_controller ||= begin
+        choices_controllers = Regexp.compile(APPLICATION_CHOICE_CONTROLLER_PATHS.join('|'))
+
+        if controller_path.match?(choices_controllers)
+          true
+        elsif controller_path.match('candidate_interface/guidance')
+          request.referer&.match?('choices')
```

Comment:
Is it not better to use?

```ruby
controller_path.match?(choices_controllers) || (controller_path.match('candidate_interface/guidance') && request.referer&.match?('choices'))
```

I always find this `returns true if check?` and then later `return true if another_check?` a little redundant.  At least the second if, ee can do in one statement.

Reply:
@inulty-dfe and I were testing different versions and found this one a bit less readable. I'm happy with either one. @inulty-dfe thoughts?

Reply:
I added a ternary operator, but on two lines as the line was becoming too long for my liking.